Add doc for NavigationServer, NavigationAgent and RVO agent avoidance (#6049)
godotengine/godot-docs: tutorials/navigation/navigation_using_agent_avoidance.rst

.. _doc_navigation_using_agent_avoidance:

Using Agent Avoidance
=====================

.. warning::

>     This chapter is a stub for linking and unfinished.

Reviewer: See my note about this warning in https://github.com/godotengine/godot-docs/pull/6048

Author: removed

This section is about how to use agent avoidance with the NavigationServer and 

> about technical detail how agent avoidance is implemented in Godot.

Reviewer: ```suggestion
documents how agent avoidance is implemented in Godot.
```

> For avoidance use with NavigationAgents Nodes see :ref:`doc_navigation_using_navigationagents`.

Reviewer: ```suggestion
For avoidance with NavigationAgents see :ref:`doc_navigation_using_navigationagents`.
```

Agent avoidance helps to prevent direct collision with other agents or moving obstacles
while the agents still follow their original velocity as best as possible.

> 

Reviewer: ```suggestion
```

> Avoidance in Godot is implemented with the help of the RVO library (Reciprocal Velocity Obstacle).

Reviewer: Non-specialist question: Isn't `Reciprocal Velocity Obstacle` more like the approach/algorithm, and we use **a** RVO library? Or is it really **the** RVO lib?

Author: Quote from their page:

> The main difference between RVO2 Library and RVO Library 1.x is the local collision avoidance technique used. RVO2 Library uses [Optimal Reciprocal Collision Avoidance](http://gamma.cs.unc.edu/CA/) (ORCA), whereas RVO Library 1.x uses [Reciprocal Velocity Obstacles](http://gamma.cs.unc.edu/RVO/) (RVO). For legacy reasons, and since both techniques are based on the same principles of reciprocal collision avoidance and relative velocity, we did not change the name of the library.

RVO places agents on a flat RVO map and gives each agent a ``radius`` and a ``position``.
Agents with overlapping radius compute a ``safe_velocity`` from their
current ``velocity``. The ``safe_velocity`` then needs to replace the original 
submitted ``velocity`` to move the actor behind the agent with custom movement code.

.. note::

    RVO avoidance is not involved in regular pathfinding, it is a completely separate system.

>     If used inappropriately the RVO avoidance can actively harm the perceived pathfinding quality.

Reviewer: ```suggestion
    If used inappropriately, RVO avoidance can actively harm the perceived pathfinding quality.
```
